src/main/java/hu/bme/mit/spaceship/TorpedoStore.java

```diff
@@ -12,6 +13,9 @@
   // rate of failing to fire torpedos [0.0, 1.0]
   private double FAILURE_RATE = 0.0; //NOSONAR
 
+  //creating a random object to be reused
+  private Random rand = new SecureRandom();
```

Comment:
Using SecureRandom() instead of  Random() and importing java.security.SecureRandom configured correctly.